src/seismometer/controls/decorators.py

```diff
@@ -2,13 +2,79 @@
 """
 As for all of the utils subpackage, no other custom subpackages should be referenced.
 """
-import builtins
+import hashlib
+import logging
+import os
+import shutil
 from functools import wraps
+from inspect import signature
+from pathlib import Path
 from typing import Any, Callable
 
-from IPython.display import display
+from IPython.display import HTML
 from ipywidgets import Widget
 
+logger = logging.getLogger("seismometer")
+SEISMOMETER_CACHE_DIR = Path("./.seismometer_cache")
+SEISMOMETER_CACHE_ENABLED = os.getenv("SEISMOMETER_CACHE_ENABLED", "") != ""
```

Comment:
Can you think of a good place to document how this bit works?  
I think there is some intuition/expectation setting we need around _when_ this occurs, since my read of this is that it's evaluated once at import.

Reply:
I hopefully added it to the sphinx docs under Low Level Patterns.